File: x2024/c/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);
    int quarters = calculate_quarters(cents);
    cents = cents - (quarters * 25);
    int dimes = calculate_dimes(cents);
    cents = cents - (dimes * 10);
    int nickels = calculate_nickels(cents);
    cents = cents - (nickels * 5);
    int pennies = calculate_pennies(cents);
    int change = quarters + dimes + nickels + pennies;
    printf("%i\n", change);
}

int calculate_quarters(int cents)
{
    // Calculating Quarters
    int quarters = 0;
    while (cents >= 25)
    {
        quarters++;
        cents = cents - 25;
    }
    return quarters;
}

int calculate_dimes(int cents)
{
    // Calculating dimes
    int dimes = 0;
    while (cents >= 10 && cents < 25)
    {
        dimes++;
        cents = cents - 10;
    }
    return dimes;
}

int calculate_nickels(int cents)
{
    // Calculating nickels
    int nickels = 0;
    while (cents >= 5 && cents < 10)
    {
        nickels++;
        cents = cents - 5;
    }
    return nickels;
}

int calculate_pennies(int cents)
{
    // Calculatig pennies
    int pennies = 0;
    while (cents >= 1 && cents < 5)
    {
        pennies++;
        cents = cents - 1;
    }
    return pennies;
}

File: x2024/python/cash/cash.py
from cs50 import get_float


def main():

    dollars = get_dollar()
    cents = int(dollars*100)

    quarters = calculate_quarters(cents)
    cents = cents - int(quarters*25)

    dimes = calc_dimes(cents)
    cents = cents - int(dimes*10)

    nickels = calc_nickels(cents)
    cents = cents - int(nickels*5)

    pennies = calc_pennies(cents)
    cents = cents - int(pennies)

    coins = quarters + dimes + nickels + pennies
    print(coins)

# function for taking input


def get_dollar():
    while True:
        dollar = get_float("Change: ")
        if dollar >= 0:
            return dollar


# function to calculate quarters


def calculate_quarters(dollars):
    quarters = int(dollars/25)
    return quarters

# function to calculate dimes


def calc_dimes(dollars):
    dimes = int(dollars/10)
    return dimes

# function to calculate nickels


def calc_nickels(dollars):
    nickels = int(dollars/5)
    return nickels

# function to calculate pennies


def calc_pennies(dollars):
    pennies = int(dollars/1)
    return pennies


main()
